app/controllers/character_weapons_controller.rb

```diff
@@ -0,0 +1,74 @@
+class CharacterWeaponsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_character_weapon, only: [:destroy, :show, :edit, :update]
+  before_action :verify_character_weapon, only: [:destroy, :show, :edit, :update]
```

Comment:
It seems both of these methods are being called on every action except `new`, change this to

```ruby
before_action :set_character_weapon, except: :new
before_action :verify_character_weapon, except: :new 
```